### ISSUE TITLE
Failed to backup a huge folder which contains numourous small files, reason : signal : killed

### ISSUE THREAD

**Issue by LeoQuote** [stashed/stash#1104]
Hi, I failed to backup nexus blobs folder, here's the stash sidecar log:

```
I0515 04:27:27.794569       1 commands.go:100] Backing up target data
[golang-sh]$ /bin/restic_0.9.6 backup /nexus-data/blobs --quiet --json --host host-0 --cache-dir /tmp/restic-cache --cleanup-cache
W0515 04:35:20.583950       1 backupsession.go:479] Failed to take backup for BackupSession deploy-sa-common-qaci-nexus-1589516841. Reason: signal: killed
I0515 04:35:20.603612       1 status.go:104] Updating status of BackupSession: sa-common/deploy-sa-common-qaci-nexus-1589516841 for host: host-0
I0515 04:35:20.681729       1 backupsession.go:184] Sync/Add/Update for Backup Session deploy-sa-common-qaci-nexus-1589516841
I0515 04:35:20.698738       1 backupsession.go:214] Skip processing BackupSession sa-common/deploy-sa-common-qaci-nexus-1589516841. Reason: BackupSession has been processed already for host "host-0"
I0515 04:35:20.698758       1 backupsession.go:184] Sync/Add/Update for Backup Session deploy-sa-common-qaci-nexus-1589516841
I0515 04:35:20.700827       1 backupsession.go:214] Skip processing BackupSession sa-common/deploy-sa-common-qaci-nexus-1589516841. Reason: BackupSession has been processed already for host "host-0"
```

stash version : v0.9.0-rc.6

I think it could be there're too many files in the folder, cause backup for other deployment are all successful, this folder is 50GB large.

Where can I get more detailed log so I can know what is the exact reason?

**Comment by hossainemruz** [comment]
Could be a cache size limit issue. Can  you try disabling caching?
https://stash.run/docs/v0.9.0-rc.6/concepts/crds/backupconfiguration/#spectempdir

**Comment by whereisaaron** [comment]
Could also be RAM, if 256MB is the default. We’d normally run 1-2GB RAM for restic with lots of files.

**Comment by LeoQuote** [comment]
> 
> 
> Could be a cache size limit issue. Can you try disabling caching?
> https://stash.run/docs/v0.9.0-rc.6/concepts/crds/backupconfiguration/#spectempdir

I tried disable cache , but no work, it's still failling

**Comment by hossainemruz** [comment]
>I0519 04:00:34.031604       1 backupsession.go:214] Skip processing BackupSession sa-common/deploy-sa-common-qaci-nexus-1589830457. Reason: BackupSession has been processed already for host "host-0"

This line is saying that backup was taken by another pod. Can you check log from other pods?

**Comment by LeoQuote** [comment]
@hossainemruz Thanks for your reply ! I've checked the deployment , its replica was set to `1`, which means there should not be another pod taking the backup.

**Comment by LeoQuote** [comment]
**Latest update**: I **set the limits to 2G** and finally it works! **the memory limit is the key**.

And previously I used the stash plugin to trigger backup, but no work, I used the cronjob to trigger backup, and it worked. perhaps the reason is I'm using an outdated stash plugin, If the problem still exist after I upgrade plugin ,I'll submit another issue.

And also, Is there any guidelines to guide us how much memory should we allocate for stash, like 1000+ -> 2G or so?
